addons/mail/static/src/core/common/store_service.js

```diff
@@ -149,6 +150,26 @@ export class Store extends BaseStore {
         };
     }
 
+    isNotificationPermissionDismissed = Record.attr(false, {
+        compute() {
+            return (
+                browser.localStorage.getItem("mail.user_setting.push_notification_dismissed") ===
+                "true"
+            );
+        },
+        /** @this {import("models").DiscussApp} */
+        onUpdate() {
+            if (this.isNotificationPermissionDismissed) {
+                browser.localStorage.setItem(
+                    "mail.user_setting.push_notification_dismissed",
```

Comment:
we might want to check who is the last login user from localStorage as well, otherwise if one user dismisses the setting, the rest of the login-in users who use the same browser won't see the notif forever

Reply:
People rarely share machines and browser sessions. Also this is a hint for suggestion, so if they don't see it anymore they can still change permissions as it's usually done in browser (or PWA).

I wouldn't over-engineer a solution that stores the session ids that have dismissed the push notification, it's more prone to errors. Besides the PWA service which contributes also to suggestion installing the PWA in the messaging menu is heavily engineered but also doesn't manage the showing of suggestion based on session id.